src/utils/debug/texture-selector-utils.ts

```diff
@@ -22,7 +22,31 @@ export function selectOriginalTextureForTileset() {
   }
 }
 
-export function selectDebugTextureForTile(tile, uvDebugTexture) {
+const cropTexture = async (texture, width, height) => {
+  if (!width || !height) {
+    return texture;
+  }
+  const bitmap = await createImageBitmap(texture);
+
+  const coeffWidth = texture.width / width;
+  const coeffHeight = texture.height / height;
+  const coeff = coeffWidth < coeffHeight ? coeffWidth : coeffHeight;
+  const sw = width * coeff;
+  const sh = height * coeff;
+
+  const canvas = document.createElement("canvas");
+  canvas.width = sw;
+  canvas.height = sh;
+  const ctx = canvas.getContext("2d");
+  if (!ctx) {
+    return null;
+  }
+
+  ctx.drawImage(bitmap, 0, 0, sw, sh);
+  return canvas;
+};
+
+export async function selectDebugTextureForTile(tile, uvDebugTexture) {
```

Comment:
Could you add TSDoc comments and types for the function? It was ported from JS long time ago and types were omited

Reply:
Done